web/modules/custom/dpl_event/src/EventSubscriber/EventSeriesRedirect.php

```diff
@@ -31,7 +36,14 @@ public function __construct(
    */
   public static function getSubscribedEvents(): array {
     return [
-      KernelEvents::REQUEST => 'checkEventSeriesRedirect',
+      KernelEvents::REQUEST => [
+        ['checkEventSeriesRedirect'],
+      ],
+      // Only target users that end up on an unpublished eventinstance page,
+      // without being allowed to see it.
+      KernelEvents::EXCEPTION => [
```

Comment:
Consider whether this is is the proper event to react to.

As I read it our listener will react to any exception being thrown within the request. I would have excepted it to be able to react on requests instead. If that is not the case and access control leading to an exception runs first then perhaps we should look into ordering?

Reply:
>If that is not the case and access control leading to an exception runs first then perhaps we should look into ordering?

I did, and no matter what I did, I couldnt catch the request before the exception ran. However, I do actually prefer it this way, as that way, the editors can still preview unpublished instances.